src/jit/assertionprop.cpp

```diff
@@ -1973,25 +1973,18 @@ AssertionInfo Compiler::optAssertionGenJtrue(GenTree* tree)
 
     GenTreeCall* call = op1->AsCall();
 
+    // See 3712, should we add chkcast...? also what about R2R
```

Comment:
What is "3712" in reference to?

Reply:
A left-behind reminder to myself to understand why the helper call lists at lines 1977 and 3712 differ, and why neither includes any R2R helpers.

Reply:
Updated this comment to describe why the code is as it is.